[v11] missing `style` package export for webpack based asset pipelines

## Describe the bug
Hello, I'm trying to upgrade a working app using shakapacker (a pre-processor based on webpack) to Choices v11 but I'm getting an error related to scss import

When updating from Choices.js 10 to Choices.js 11, shakapacker is not able
to resolve
- `@import "choices.js/src/styles/choices";` in Scss
- `import "choices.js/public/assets/styles/choices.css";` in JS

This is due to the new `exports` entry in `package.json`, which require
specific entries for webpack, such as `style`

## To Reproduce
### Tools and versions required to replicate

- npm 10.8.0
- Shakapacker 7/8
- Node 20.15.0
- Ruby 3.3.4 (should not be relevant)
- Rails 7.2.1 (anything >= 6.1.0 should not be relevant)

### Initial app
```sh
rails _7.2.1_ new test-choices-package-exports --skip-active-record --skip-javascript --skip-git --skip-test
cd test-choices-package-exports
echo "{ \"private\": true, \"packageManager\": \"npm@10.8.2\" }" > package.json
echo "gem 'shakapacker', '8.0.2'" >> Gemfile
bundle
rails shakapacker:install
npm add css-loader style-loader mini-css-extract-plugin css-minimizer-webpack-plugin
npm add choices.js@10.2.0
echo "module.exports = { plugins: [require('postcss-import')] }" > postcss.config.js
git add .
git commit -am "First commit"
git branch tests/scss main
git branch tests/css main
```

#### CSS import from JavaScript
```
git checkout tests/css
echo "import \"choices.js/public/assets/styles/choices.css\";" >> app/javascript/packs/application.js
git commit -am "CSS test"
bin/shakapacker # WORKS, CSS OUTPUT PRESENT
npm add choices.js@^11.0.0
bin/shakapacker # DOES NOT WORK
```

Error:
```
ERROR in ./app/javascript/packs/application.js 16:0-53
Module not found: Error: Package path ./public/assets/styles/choices.css is not exported from package ./node_modules/choices.js (see exports field in ./node_modules/choices.js/package.json)
```

#### Sass

```
git checkout tests/scss
npm add sass-loader sass
echo "@import \"choices.js/src/styles/choices.scss\";" >> app/javascript/packs/application.scss
git add .
git commit -am "Scss test"
bin/shakapacker # WORKS, CSS OUTPUT PRESENT
npm add choices.js@^11.0.0
bin/shakapacker # DOES NOT WORK
```

Error:
```
Error: Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
Can't find stylesheet to import.
  ╷
1 │ @import "choices.js/src/styles/choices.scss";
  │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  app/javascript/packs/application.scss 1:9  root stylesheet
```

### Suggested solution

https://github.com/ifad/Choices/tree/bugfix/style-export

1. `npm add choices.js@ifad/choices#bugfix/style-export`
2. Retry `bin/shakapacker`

Details are in the commit message

Ref: https://webpack.js.org/guides/package-exports/#reference-syntax

--- a/package.json
+++ b/package.json
@@ -12,7 +12,9 @@
     ".": {
       "types": "./public/types/src/index.d.ts",
       "import": "./public/assets/scripts/choices.mjs",
-      "require": "./public/assets/scripts/choices.js"
+      "require": "./public/assets/scripts/choices.js",
+      "style": "./public/assets/styles/choices.css",
+      "sass": "./src/styles/choices.scss"
     },
     "./search-basic": {
       "types": "./public/types/src/index.d.ts",
@@ -23,9 +25,15 @@
       "types": "./public/types/src/index.d.ts",
       "import": "./public/assets/scripts/choices.search-none.mjs",
       "require": "./public/assets/scripts/choices.search-none.min.js"
-    }
+    },
+    "./public/assets/styles/*.css": "./public/assets/styles/*.css",
+    "./src/styles/*.scss": "./src/styles/*.scss",
+    "./src/styles/*": "./src/styles/*.scss"
   },
-  "sideEffects": false,
+  "sideEffects": [
+    "*.scss",
+    "*.css"
+  ],
   "scripts": {
     "start": "run-p js:watch css:watch",
     "build": "run-p js:build css:build",

